### ISSUE TITLE
integrate https://github.com/LokiResearch/fast-triangle-triangle-intersection into triangle utils

### ISSUE THREAD

**Issue by coolvision** [gkjohnson/three-mesh-bvh#496]
WDYT about integrating fast-triangle-triangle-intersection into triangle utils?

I haven't done a benchmark yet, so not sure if it's faster, but at least one improvement would be handling of coplanar triangles (which is currently TODO in intersectsTriangle).

It could be used instead of ExtendedTriangle.prototype.intersectsTriangle, but the problem is that in case of coplanar triangles return value might be 1-3 points instead of a single edge. So maybe it could be an additional function, like _intersectsTriangleAllPoints_.

I would be happy to do a pull request if this is something that seems useful.

**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by gkjohnson** [comment]
Hello! What's the motivation here? This was originally suggested in #384 and if I recall the function was slower than the current implementation, though I agree there are some issues to fix in the current implementation. It's also possible the implementation has improved.

The original intent behind providing a triangle-triangle-intersection function  was to enable the [three-bvh-csg](https://github.com/gkjohnson/three-bvh-csg) library. I haven't put a lot of effort into it recently but I would like to keep the intersection function internal until I can invest time in the CSG project to hone the behavior of the intersection function accordingly.

In the mean time if you need a more sophisticated tri-tri intersect function behavior the BVH class is flexible enough to afford this using the `shapecast` function.

**Comment by coolvision** [comment]
yes, initially I actually started to test fast-triangle-triangle-intersection with three-bvh-csg, in hope that it would help with some corner cases.
but indeed, after more testing now I think that having coplanar intersections is not useful for CSG.
thanks for the feedback, closing this one.
